src/PaperG/FirehoundBlob/BasicInfo.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+namespace PaperG\FirehoundBlob;
+
+
+class BasicInfo implements BlobInterface
+{
+    const NAME = "name";
+    const UUID = "uuid";
+    const METADATA = "metadata";
+    const SCENARIO = "scenario";
+    const BLOB = "blob";
+
+    /**
+     * @var string Advertising campaign name
+     */
+    private $name;
+
+    /**
+     * @var string identifier
+     */
+    private $uuid;
+
+    /**
+     * @var string purpose of sending this blob
+     */
+    private $metadata;
+
+    /**
+     * @var string
+     */
+    private $scenario;
+
+    /**
+     * @var BlobInterface
+     */
+    private $blob;
+
+    public function __construct() {}
+
+    /**
+     * @param string $metadata
+     */
+    public function setMetadata($metadata)
+    {
+        $this->metadata = $metadata;
+    }
+
+    /**
+     * @return string
+     */
+    public function getMetadata()
+    {
+        return $this->metadata;
+    }
+
+    /**
+     * @param string $name
+     */
+    public function setName($name)
+    {
+        $this->name = $name;
+    }
+
+    /**
+     * @return string
+     */
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    /**
+     * @param string $scenario
+     */
+    public function setScenario($scenario)
+    {
+        $this->scenario = $scenario;
+    }
+
+    /**
+     * @return string
+     */
+    public function getScenario()
+    {
+        return $this->scenario;
+    }
+
+    /**
+     * @param BlobInterface $subBlob
+     */
+    public function setBlob($subBlob)
+    {
+        $this->blob = $subBlob;
+    }
+
+    /**
+     * @return BlobInterface
+     */
+    public function getBlob()
+    {
+        return $this->blob;
+    }
+
+    /**
+     * @param string $uuid
+     */
+    public function setUuid($uuid)
+    {
+        $this->uuid = $uuid;
+    }
+
+    /**
+     * @return string
+     */
+    public function getUuid()
+    {
+        return $this->uuid;
+    }
+
+    public function toArray()
+    {
+        return [
+            self::NAME => $this->getName(),
+            self::UUID => $this->getUuid(),
+            self::METADATA => $this->getMetadata(),
+            self::SCENARIO => $this->getScenario(),
+            self::BLOB => isset($this->blob) ? $this->getBlob()->toArray() : null
+        ];
+    }
+
+    public function fromArray($array)
+    {
+        $this->setName($this->safeGet($array, self::NAME));
+        $this->setUuid($this->safeGet($array, self::UUID));
+        $this->setMetadata($this->safeGet($array, self::METADATA));
+        $this->setScenario($this->safeGet($array, self::SCENARIO));
+        $this->setBlob($this->safeGet($array, self::BLOB));
+    }
+
+    private function safeGet($array, $key)
+    {
+        return isset($array[$key]) ? $array[$key] : null;
+    }
+
+    public function validate()
```

Comment:
## Suggestion 1: Validate as a class
I suggest breaking `validate` out into a separate class, say `BlobValidator`, that knows how to validate a blob accordingly. It can validate a basic blob, a FB unmanaged or AN-specific blob. The class knows how to run validation, and we don't care about how the validation is done within the blob.

Within the blob validator class, we can then use composition to include different types of validators. A BasicBlobValidator implements `validate`, a FBBlobValidator also implements `validate` etc. And these validator classes are passed into the BlobValidator class. The BlobValidator class will then call validateBasicBlob($blob), validateFbBlob($blob) or validateAnDesktopBlob($blob). 

Code-wise it'll look like this:

```php
$blob = new BasicInfo($json);
$validator = new BlobValidator();

if ($validator->validateBasic($blob)) { // do something; }

$scenario = $blob->getScenario();
switch ($scenario) {
   case ("Unmanaged-Facebook"): 
      $scenarioBlob = new FacebookInfoBlob($blob->getBlob());
      if ( $validator->validateFbInfoBlob($scenarioBlob) ) { // do something }
   }
}
```

## Suggestion 2: Validate as a method
Having validate as a method is clean and removes the hassle of having to initialize a separate validator class. However, the downside is that if we do decide to have multiple versions of scenario blobs in the future, then the validate method might get really long as it needs to run different checks depending on the version of the blob. 

I like the idea of this:

```php
$blob = new BasicInfo($json);

if ($blob->validate()) {
   $scenario = $blob->getScenario();
   switch ($scenario) {
      case "Unmanaged-Facebook": 
         $scenarioBlob = new FbInfoBlob($blob->getBlob());
         if ($scenarioBlob->validate()) { //do something }
   }
```